madanalysis/job/job_tagger_header.py

```diff
@@ -1,67 +1,100 @@
+        file.write('#define MADANALYSIS5_NEW_TAGGER_H\n')
         file.write('// SampleAnalyzer headers\n')
-        file.write('#include "SampleAnalyzer/Commons/DataFormat/EventFormat.h"\n')
-        file.write('#include "SampleAnalyzer/Commons/DataFormat/SampleFormat.h"\n')
-        file.write('namespace MA5\n')
-        file.write('{\n')
-        file.write('  class NewTagger\n')
-        file.write('  {\n')
+        file.write('#include "SampleAnalyzer/Commons/Base/SFSTaggerBase.h"\n')
+        file.write('namespace MA5 {\n')
+        file.write('    class NewTagger: public SFSTaggerBase {\n')
         file.write('    public :\n')
-        file.write('      /// Constructor without argument\n')
-        file.write('      NewTagger() \n')
-        file.write('      {\n')
-        if not self.fastsim.isNewSmearerOn():
-            # If tagger is on but SmearerBase wont be initialized, write SFS reference
-            # This way whenever SFS is used it will consistently show the ref.
-            file.write('        INFO <<"   <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>" << endmsg;\n')
-            file.write('        INFO <<"   <>                                                              <>" << endmsg;\n')
-            file.write('        INFO <<"   <>     Simplified Fast Detector Simulation in MadAnalysis 5     <>" << endmsg;\n')
-            file.write('        INFO <<"   <>            Please cite arXiv:2006.09387 [hep-ph]             <>" << endmsg;\n')
-            file.write('        INFO <<"   <>                                                              <>" << endmsg;\n')
-            file.write('        INFO <<"   <>         https://madanalysis.irmp.ucl.ac.be/wiki/SFS          <>" << endmsg;\n')
-            file.write('        INFO <<"   <>                                                              <>" << endmsg;\n')
-            file.write('        INFO <<"   <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>" << endmsg;\n')
-        file.write('      }\n\n')
-        file.write('      /// Destructor\n')
-        file.write('      virtual ~NewTagger() {}\n\n')
-        file.write('      /// Tagger execution\n')
-        file.write('      void Execute(SampleFormat &mySample, EventFormat &myEvent);\n\n')
-        file.write('  };\n')
-        file.write('}\n')
-        file.write('#endif')
+        file.write('        /// Constructor without argument\n')
+        file.write('        NewTagger() {}\n\n')
+        file.write('        /// Destructor\n')
+        file.write('        virtual ~NewTagger() {}\n\n')
+        file.write('        void Initialize() {\n')
+        file.write('            /// @brief Booleans for code efficiency\n')
+        file.write('            /// Turn on the usage of tau tagging efficiency\n')
+        file.write(f"            _isTauTaggingEffOn = {'true' if (15, 15) in [(x[0], x[1]) for x in self.unique_rules] else 'false'};\n")
```

Comment:
What happens if only a mistagging (jets as tau for instance) rate is inputed?

Reply:
These booleans correspond to the efficiencies which have been implemented in the code; not all portions have efficiency requirements/possibilities, so booleans are there only for the if conditions that are implemented in the code. (assuming that I didn't miss any)

Reply:
Self Note: revert this